rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -217,6 +220,32 @@ private void initializeWebClient() throws RestClientException {
         webClient = builder.baseUrl(config.getBaseUrl()).clientConnector(connector).build();
     }
 
+    /**
+     * Injects a traceparent header into the provided HttpHeaders object.
+     * The header is composed of the trace ID and span ID obtained from the current SpanContext.
+     * The header is only added if both trace ID and span ID are present.
+     *
+     * @param headers The HttpHeaders object where the traceparent header will be added.
+     *                This object is expected to be non-null.
+     */
+    private static void injectTraceparentHeader(final HttpHeaders headers) {
+        final Optional<SpanContext> spanContext = Optional.ofNullable(Span.current().getSpanContext());
+
+        if (spanContext.isPresent() && headers != null) {
+            final String traceId = spanContext.map(SpanContext::getTraceId).orElse(null);
+            final String spanId = spanContext.map(SpanContext::getSpanId).orElse(null);
+            final TraceFlags traceFlags = spanContext.map(SpanContext::getTraceFlags).orElse(null);
```

Comment:
Why so complicated solution with `Optional`?
```suggestion
        if (headers == null) {
            return;
        }

        final SpanContext spanContext = Span.current().getSpanContext();
        
        if (spanContext != null) {
            final String traceId = spanContext.getTraceId();
            final String spanId = spanContext.getSpanId();
            final TraceFlags traceFlags = spanContext.getTraceFlags();Flags = spanContext.map(SpanContext::getTraceFlags).orElse(null);
```